internal/util/metautil/segment_index.go

```diff
@@ -0,0 +1,26 @@
+package metautil
+
+import (
+	"path"
+
+	"github.com/milvus-io/milvus/internal/common"
+)
+
+func BuildSegmentIndexFilePath(rootPath string, buildID, indexVersion, partID, segID int64, fileKey string) string {
+	k := JoinIDPath(buildID, indexVersion, partID, segID)
+	return path.Join(rootPath, common.SegmentIndexPath, k, fileKey)
+}
+
+func BuildSegmentIndexFilePathWithPrefix(prefix string, buildID, indexVersion, partID, segID int64, fileKey string) string {
```

Comment:
What's the difference between `BuildSegmentIndexFilePathWithPrefix` and `BuildSegmentIndexFilePath`?
Should we also join `common.SegmentIndexPath` when `BuildSegmentIndexFilePathWithPrefix`?